File: Python/747.py
class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        if len(nums) <= 1:
            return 0
        maxi = max(nums)
        if maxi >= 2 * sorted(nums)[-2]:
            return nums.index(maxi)
        return -1
        

File: Solutions/Java/747.java
class Solution {
    public int dominantIndex(int[] nums) {
        int max = 0;
        int secondMax = 0;
        int maxIndex = 0;
        
        for (int i=0; i<nums.length; i++) {
            if (nums[i]>max) {
                maxIndex = i;
                secondMax = max;
                max = nums[i];
            }
            else if (nums[i]>secondMax) {
                secondMax = nums[i];
            }
        }
        if (max<2*secondMax)
                    return -1;
        return maxIndex;
    }
}
